### PR TITLE
[WIP] Add aggregations and more pretty charts to share 

### PR DESCRIPTION
**Pull request by bdyetton**

### CODE REVIEW

File: website/static/js/share/stats.js [hunk 1]
```diff
@@ -43,7 +32,7 @@ function donutGraph (data, vm) {
         tooltip: {
             format: {
                 name: function (name, ratio, id, index) {
-                    if (name === 'pubmed') {
+                    if (name === 'pubmed') { //TODO @fabianvf, can we get rid of this now? looks like pubmedcentral is already the name of one of the sources
```
Comment (fabianvf) [inline website/static/js/share/stats.js]: yes, this was a hotfix hack and can be removed.

File: website/static/js/share/stats.js [hunk 2]
```diff
@@ -92,6 +100,90 @@ function timeGraph (data) {
...
+    dateTemp.setMonth(dateTemp.getMonth() - 3);
+    var threeMonthsAgo = dateTemp.getTime();
+    var dateHistogramQuery = {'match_all':{}};
+    var dateHistogramAgg = {'sourcesByTimes': utils.termsFilter('field','_type')};
+    dateHistogramAgg.sourcesByTimes['aggregations'] = {'articlesOverTime' :
+        utils.dateHistogramFilter('providerUpdatedDateTime',threeMonthsAgo)};
+    return {'query' : dateHistogramQuery, 'aggregations': dateHistogramAgg ,'filters' : {}}
+};
+
+Stats.timeSinceEpochInMsToMMYY = function(timeSinceEpochInMs)
```
Comment (fabianvf) [inline website/static/js/share/stats.js]: I think you can also specify the return format of the dates in the aggregation itself, if you'd prefer.

File: website/static/js/share/stats.js [hunk 3]
```diff
@@ -92,6 +100,90 @@ function timeGraph (data) {
...
+    chartData.colors = {};
+    chartData.type = 'area-spline';
+    chartData.x = 'x';
+    chartData.groups = [];
+    var grouping = [];
+    grouping.push('x');
+    var hexColors = utils.generateColors(data.aggregations.sourcesByTimes.buckets.length);
+    var i = 0;
+    var datesCol = [];
+    data.aggregations.sourcesByTimes.buckets.forEach( //TODO @bdyetton what would be nice is a helper function to do this for any agg returned by elastic
```
Comment (fabianvf) [inline website/static/js/share/stats.js:144]: This aggregation should work on arbitrary datetime aggregations though, which is probably abstraction enough for now.

File: website/static/js/share/utils.js [hunk 4]
```diff
@@ -278,6 +283,25 @@ utils.boolQuery = function(must, must_not, should, minimum) {
     return ret;
 };
 
+utils.dateHistogramFilter = function(feild,gte,lte,interval){
```
Comment (fabianvf) [inline website/static/js/share/utils.js]: is this a filter or an aggregation? Also, `field`, not `feild`

File: website/static/js/share/stats.js (hunk 3 above)
Comment (bdyetton) [inline website/static/js/share/stats.js:144]: So, do i need to change anything here?

Comment (fabianvf) [inline website/static/js/share/stats.js:144]: nope, just letting you know

**Pull request merged**